src/pint/models/tcb_conversion.py

```diff
@@ -0,0 +1,168 @@
+
+    Parameters
+    ----------
+    model : pint.models.timing_model.TimingModel
+        The timing model
+    param : str
+        The parameter name to be converted
+    backwards : bool
+        Whether to do TDB to TCB conversion.
+    """
+    factor = IFTE_K if backwards else 1 / IFTE_K
+    tref = IFTE_MJD0
+
+    if hasattr(model, param) and getattr(model, param).quantity is not None:
+        par = getattr(model, param)
+        assert isinstance(par, MJDParameter)
+
+        par.value = (par.value - tref) * factor + tref
+        if par.uncertainty_value is not None:
+            par.uncertainty_value *= factor
+
+
+def convert_tcb_tdb(model, backwards=False):
+    """This function performs a partial conversion of a model
+    specified in TCB to TDB. While this should be sufficient as
+    a starting point, the resulting parameters are only approximate
+    and the model should be re-fit.
+
+    This is based on the `transform` plugin of tempo2.
+
+    The following parameters are converted to TDB:
+        1. Spin frequency, its derivatives and spin epoch
+        2. Sky coordinates, proper motion and the position epoch
+        3. DM, DM derivatives and DM epoch
+        4. Keplerian binary parameters and FB1
+
+    The following parameters are NOT converted although they are
+    in fact affected by the TCB to TDB conversion:
+        1. Parallax
+        2. TZRMJD and TZRFRQ
+        2. DMX parameters
+        3. Solar wind parameters
+        4. Binary post-Keplerian parameters including Shapiro delay
+           parameters (except FB1)
+        5. Jumps and DM Jumps
+        6. FD parameters
+        7. EQUADs
+        8. Red noise parameters including FITWAVES, powerlaw red noise and
+           powerlaw DM noise parameters
+
+    Parameters
+    ----------
+    model : pint.models.timing_model.TimingModel
+       Timing model to be converted.
+    backwards : bool
+        Whether to do TDB to TCB conversion. The default is TCB to TDB.
+    """
+
+    if model.UNITS in ["TDB", None]:
+        log.warning("The input par file is already in TDB units. Doing nothing.")
+        return
+
+    log.warning(
+        "The TCB to TDB conversion is only approximate. "
+        "The resulting timing model should be re-fit to get reliable results."
+    )
+
+    if "Spindown" in model.components:
+        for n, Fn_par in model.get_prefix_mapping("F").items():
+            scale_parameter(model, Fn_par, n + 1, backwards)
+
+        transform_mjd_parameter(model, "PEPOCH", backwards)
+
+    if "AstrometryEquatorial" in model.components:
+        scale_parameter(model, "PMRA", 1, backwards)
+        scale_parameter(model, "PMDEC", 1, backwards)
+        transform_mjd_parameter(model, "POSEPOCH", backwards)
+    elif "AstrometryEcliptic" in model.components:
+        scale_parameter(model, "PMELAT", 1, backwards)
+        scale_parameter(model, "PMELONG", 1, backwards)
+        transform_mjd_parameter(model, "POSEPOCH", backwards)
```

Comment:
Is there a reason to have the `transform_mjd_parameter(model, "POSEPOCH", backwards)` independently in each clause, rather than just run it after the `if/elif` block finishes? Just reduces repeated code (I think?).

Reply:
Resolved.